src/extensions/php-fpm-metrics/pkg/phpfpm/status.go

```diff
@@ -0,0 +1,69 @@
+package phpfpm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/nginx/agent/v2/src/core"
+	log "github.com/sirupsen/logrus"
+)
+
+var shell core.Shell = core.ExecShellCommand{}
+
+// Status is an Enum that represents the status of PhpFpm.
+type Status int
+
+const (
+	UNKNOWN Status = iota
+	INSTALLED
+	RUNNING
+)
+
+// String get the string representation of the enum
+func (s Status) String() string {
+	switch s {
+	case INSTALLED:
+		return "INSTALLED"
+	case RUNNING:
+		return "RUNNING"
+	}
+	return "UNKNOWN"
+}
+
+// GetStatus returns phpfpm process status
+func GetStatus(pid, version string) Status {
```

Comment:
recommend returning `(Status, error)`, and letting the caller decide what to do with errors. This ends up being a more maintainable approach in general, as you can add error handling without touching a broad swath of the codebase.